src/slrcost.jl

```diff
@@ -258,7 +258,21 @@ using Mimi
                     error("The `popinput` argument values of 1 and 2 are not supported at this time.  In the future they will indicate use of Jones and O'Neill 2016 or Merkens et al 2016 population data, respectively.")
                     # v.popdens_seg[t,m]=p.popdens_seg_merkens[ti1,m]
                 end
-                v.areaparams[m, :] = [p.area1[m] p.area2[m] p.area3[m] p.area4[m] p.area5[m] p.area6[m] p.area7[m] p.area8[m] p.area9[m] p.area10[m] p.area11[m] p.area12[m] p.area13[m] p.area14[m] p.area15[m]]
+                v.areaparams[m, 1] = p.area1[m]
+                v.areaparams[m, 2] = p.area2[m]
+                v.areaparams[m, 3] = p.area3[m]
+                v.areaparams[m, 4] = p.area4[m]
+                v.areaparams[m, 5] = p.area5[m]
+                v.areaparams[m, 6] = p.area6[m]
+                v.areaparams[m, 7] = p.area7[m]
+                v.areaparams[m, 8] = p.area8[m]
+                v.areaparams[m, 9] = p.area9[m]
+                v.areaparams[m, 10] = p.area10[m]
+                v.areaparams[m, 11] = p.area11[m]
+                v.areaparams[m, 12] = p.area12[m]
+                v.areaparams[m, 13] = p.area13[m]
+                v.areaparams[m, 14] = p.area14[m]
+                v.areaparams[m, 15] = p.area15[m]
```

Comment:
This can be simplified a bit with [`Base.Cartesian.@nexprs`](https://docs.julialang.org/en/v1/devdocs/cartesian/#Base.Cartesian.@nexprs) (you need to do some `Symbol` gymnastic though, thanks @seelengrab)
```suggestion
                Base.Cartesian.@nexprs 15 i -> v.areaparams[m, i] = getproperty(p, Symbol(:area, i))[m]
```

Reply:
Thanks, this is pretty cool! Having said that, I think in this code base here it is not the right solution, it is more important for us to have code that is easily understandable by folks that are not super familiar with Julia, and I think the more verbose version is better for that.